e2etests/pkg/routes/routes.go

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package routes
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net"
+
+	"go.universe.tf/e2etest/pkg/frr"
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+	"go.universe.tf/e2etest/pkg/ipfamily"
+	"go.universe.tf/metallb/e2etest/pkg/executor"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// CheckNeighborHasPrefix tells if the given frr container has a route toward the given prefix
+// via the set of node passed to this function.
+func CheckNeighborHasPrefix(neighbor frrcontainer.FRR, prefix string, nodes []v1.Node) (bool, error) {
+	routesV4, routesV6, err := frr.Routes(neighbor)
+	if err != nil {
+		return false, err
+	}
+
+	_, cidr, err := net.ParseCIDR(prefix)
+	if err != nil {
+		return false, err
+	}
+
+	route, err := routeForCIDR(cidr, routesV4, routesV6)
+	var notFound RouteNotFoundError
+	if errors.As(err, &notFound) {
+		return false, nil
+	}
```

Comment:
why is this needed?

Reply:
Because the specific error is different from a conversion error or something not recoverable.
Even though routeForCIDR can only return that type of error, it's better to handle the specific error separately from the generic one, it will make the logic future proof.